src/download.rs

```diff
@@ -549,3 +572,176 @@ mod error {
         },
     }
 }
+
+#[async_trait]
+trait SnapshotWriteTarget: AsRef<Path> {
+    // grow the target to the desired length
+    async fn grow(&mut self, length: i64) -> Result<()>;
+
+    // returns the file path to which blocks must be written
+    fn write_path(&self) -> Result<&Path>;
```

Comment:
The `AsRef<Path>` trait bound was meant to satisfy this. I'm fine with doing it this way as it's more clear and less abstract. You hit upon the main weakness of leveraging `AsRef<Path>` which is that there are two paths we might want to reference in the case of the temp file.

Let's clean up the `AsRef<Path>` trait bound and implementations since they're not used.

Reply:
Thanks. I have made the change.